autosklearn/ensemble_building/manager.py

```diff
@@ -0,0 +1,371 @@
+from __future__ import annotations
+
+from typing import Any
+
+import logging.handlers
+import time
+import traceback
+
+import dask.distributed
+import numpy as np
+from smac.callbacks import IncorporateRunResultCallback
+from smac.optimizer.smbo import SMBO
+from smac.runhistory.runhistory import RunInfo, RunValue
+from smac.tae.base import StatusType
+
+from autosklearn.automl_common.common.utils.backend import Backend
+from autosklearn.ensemble_building.builder import EnsembleBuilder
+from autosklearn.metrics import Scorer
+from autosklearn.util.logging_ import get_named_client_logger
+
+
+class EnsembleBuilderManager(IncorporateRunResultCallback):
+    def __init__(
+        self,
+        backend: Backend,
+        dataset_name: str,
+        task: int,
+        metric: Scorer,
+        time_left_for_ensembles: float = 10,
```

Comment:
I didn't see why not as it's something we can give a default to and still have correct behaviour.

I agree that it's sort of a useless value though and there's no real "good" default.

I'll remove it if it's a contention point

Reply:
I don't mind giving a default now that I think about this. But then I'd rather provide an infinite default, because otherwise the ensemble builder will just terminate ensemble selection after 10s and not return an ensemble as the default behavior.

Reply:
Will do